src/main/java/school/hei/vola/service/OrangeSyncService.java

```diff
@@ -0,0 +1,92 @@
+package school.hei.vola.service;
+
+import static school.hei.vola.model.psp.PspType.ORANGE_MONEY;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.time.LocalDate;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import school.hei.vola.endpoint.event.model.PaymentVerificationRequested;
+import school.hei.vola.model.psp.orange.OrangeApiClient;
+import school.hei.vola.model.psp.orange.OrangeTransaction;
+import school.hei.vola.repository.PaymentRepository;
+import school.hei.vola.repository.jpa.JOrangeTransactionRepository;
+import school.hei.vola.repository.jpa.model.JOrangeTransaction;
+import school.hei.vola.service.event.PaymentVerificationRequestedService;
+import school.hei.vola.service.sync.model.RecoveryResult;
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class OrangeSyncService {
+  private final OrangeApiClient orangeApiClient;
+  private final JOrangeTransactionRepository transactionRepository;
+  private final PaymentRepository paymentRepository;
+  private final PaymentVerificationRequestedService verificationService;
+
+  public RecoveryResult sync(LocalDate date) {
+    try {
+      var transactions = orangeApiClient.transactionsOf(date).getTransactions();
+      int totalTransactions = transactions.size();
+      int syncedTransactions =
+          transactions.stream()
+              // 1 if persisted & verified, else 0
+              .mapToInt(tx -> persistAndVerifyTransaction(tx) ? 1 : 0)
+              .sum();
+
+      return RecoveryResult.builder()
+          .date(date)
+          .isSuccessful(true)
+          .nbSyncedTransactions(syncedTransactions)
+          .nbTotalTransactions(totalTransactions)
+          .build();
+    } catch (Exception e) {
+      log.error("[SYNC] Failed to sync transactions for date={}", date, e);
+      return RecoveryResult.builder()
+          .date(date)
+          .isSuccessful(false)
+          .nbSyncedTransactions(0)
+          .nbTotalTransactions(0)
+          .errorMessage(e.getMessage())
+          .build();
+    }
+  }
+
+  private boolean persistAndVerifyTransaction(OrangeTransaction tx) {
+    try {
+      boolean inserted = persistIfNew(tx);
+      triggerVerificationIfExists(tx);
+      return inserted;
+    } catch (Exception e) {
+      log.error(
+          "[SYNC] Failed to sync transaction ref={}, amount={}, details={}",
```

Comment:
"[SYNC] Failed to sync orange transaction ot={}, tx" is vastly superior, with appropriate ::toString on OrangeTransaction.

Furthermore ot instead of tx is better.